api/services/images/v1/images.proto

```diff
@@ -89,6 +89,8 @@ message GetImageResponse {
 
 message CreateImageRequest {
 	Image image = 1;
+
+	google.protobuf.Timestamp source_date_epoch = 2;
```

Comment:
@dmcgowan @tonistiigi 

Let me know if these `source_date_epoch` properties  (also present in [`diff.proto`](https://github.com/containerd/containerd/pull/7478/files#diff-7acc182342961784b44f85f2067a1127963b3d0b2775c59f4240333e2d14f1eeR79)) should still remain in the proto message definitions, or should be moved to the grpc header as in the `containerd-namespace`.

Reply:
The latter would be simpler, but needs partially reverting:
- https://github.com/containerd/containerd/pull/7478 (merged in v1.7.0-beta.0)

Reply:
I think they should be in proto. See also https://github.com/moby/buildkit/issues/3180 for differ side.

Reply:
POC for moving the proto to the grpc header: https://github.com/AkihiroSuda/containerd/commit/c2c6b289d7e6759243000b12a8cc56703720566d
This seems much simpler.

- - -
@tonistiigi 

> I think they should be in proto.

Could you elaborate the reason?

Reply:
It's part of the parameters for the API request. Shouldn't be hidden in a side-channel.

Reply:
True, but sidechanneling significantly simplifies the code, and actually we have been already sidechanneling the `containerd-namespace`

Reply:
> and actually we have been already sidechanneling the containerd-namespace

fwiw I think that is a mistake as well. But at least that one is a parameter to most(maybe all) request. While this field only carries meaning for very specific requests.

For the differ, we will eventually need two timestamps probably. One for the whiteout behavior that exists currently but also one to control the behavior for new files.

Reply:
+1 for having an explicit field.

Reply:
Agreed this is not a good use case for grpc headers, disagree that namespaces is not 😉